### PR TITLE
Added 404 fallback page

### PR DESCRIPTION
**Pull request by 1Luc1**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #192 

no cats, sry :smile_cat:

### CODE REVIEW

File: views/site/404.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+use Yiisoft\Html\Html;
+?>
+
+<div class="card shadow p-5 my-5 mx-5 bg-white rounded">
+    <div class="card-body text-center ">
+        <h1 class="card-title display-1 fw-bold">404</h1>
+        <p class="card-text">
+            <?php echo "The page "
+                . Html::span($requestedUri, ['class' => 'text-muted'])
```
Comment (Mister-42) [inline views/site/404.php:18]:
```php
. Html::span($urlMatcher->getCurrentUri()->getPath(), ['class' => 'text-muted'])`
```

File: src/Handler/NotFoundHandler.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    public function __construct(ViewRenderer $viewRenderer)
+    {
+        $this->viewRenderer = $viewRenderer->withControllerName('site');
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        return $this->viewRenderer
+            ->render('404', ['requestedUri' => (string) $request->getUri()])
```
Comment (Mister-42) [inline src/Handler/NotFoundHandler.php:25]: $urlMatcher is available in view, so no need to add parameter to view.

File: views/site/404.php (hunk 1 above)
Comment (samdark) [reply views/site/404.php:18]: $requestedUri should be passed through `Html::encode()`.

Comment (Mister-42) [reply views/site/404.php:18]: > $requestedUri should be passed through `Html::encode()`.

I changed my proposal accordingly 👍🏻

**Comment by Mister-42** [comment]
Hard way to learn `$urlMatcher->getCurrentRoute()` will be `NULL`. Makes sense, but something to be aware of when used in layout.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**